### PR TITLE
Add directive for consistent file inputs across browsers

### LINKED ISSUE #534: Importing the same exercise twice does not work
When importing the same file twice without reloading the page in between, the second import has no noticeable effect.

### PR DESCRIPTION
**Pull request by Dassderdie**
This fixes #534. It uses the same workaround as #541 but encapsulates it in a reusable directive instead of directly doing it in the component.

### CODE REVIEW

File: frontend/src/app/shared/directives/file-input.directive.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * @example
+ * ````html
+ * <input type="file" (appFileInput)="processFileList($event)" />
+ * ````
+ */
+@Directive({
+    selector: 'input[appFileInput]',
+})
+export class FileInputDirective implements OnDestroy {
+    destroy$ = new Subject<void>();
```
Comment (Nils1729) [inline frontend/src/app/shared/directives/file-input.directive.ts:18]: Dust
Is the `destroy$` member required here? I could not find any observers to it.

Comment (Dassderdie) [reply frontend/src/app/shared/directives/file-input.directive.ts:18]: Very good observation!
I'll look into it this evening.

Comment (Dassderdie) [reply frontend/src/app/shared/directives/file-input.directive.ts]: I removed the (unused) destroy logic in dced4d474d0dc8d697386b23f6c801266f56b7f9.

I believe it is safe to assume that `@HostListener` automatically unsubscribes and, while `appFileInput` doesn't `complete()` when the directive is destroyed, angular automatically unsubscribes from Outputs in templates.

So there should hopefully be no memory leak.

**Review by Nils1729 — APPROVED** [review]
LGTM

**Pull request merged**